src/fides/api/service/connectors/website_connector.py

```diff
@@ -0,0 +1,81 @@
+from typing import Any, Dict, List, Optional
+
+from loguru import logger
+from requests import head
+
+from fides.api.common_exceptions import ConnectionException
+from fides.api.graph.execution import ExecutionNode
+from fides.api.models.connectionconfig import ConnectionTestStatus
+from fides.api.models.policy import Policy
+from fides.api.models.privacy_request import PrivacyRequest, RequestTask
+from fides.api.service.connectors.base_connector import BaseConnector
+from fides.api.service.connectors.query_configs.query_config import QueryConfig
+from fides.api.util.collection_util import Row
+
+
+class WebsiteConnector(BaseConnector):
+    """
+    Website connector, used currently for Website 'monitoring' - this is class is used to test basic connections to the website
+
+    NOTE: No DSR processing is supported for Website connectors.
+    """
+
+    def create_client(self) -> Any:  # type: ignore
+        """Returns a client for the website"""
+
+    def query_config(self, node: ExecutionNode) -> QueryConfig[Any]:
+        """DSR execution not supported for Website connectors"""
+        raise NotImplementedError()
+
+    def test_connection(self) -> Optional[ConnectionTestStatus]:
+        """
+        Validates the connection to the website by executing a `HEAD` request against the provided URL.
+
+        TODO: can we perform a better validation by pinging the website from the web monitor proxy?
+        TODO: can we validate credentials somehow when we support them?
+        """
+        website_url = self.configuration.secrets.get("url")
+        logger.info(
+            "Starting test connection to connector '{}', at URL '{}'",
+            self.configuration.key,
+            website_url,
+        )
+        try:
+            response = head(self.configuration.secrets["url"])
+        except Exception as error:
+            raise ConnectionException(str(error))
+
+        if response.status_code >= 400:
+            raise ConnectionException(
+                f"HEAD request to '{website_url}' resulted in error status code: '{response.status_code}'"
```

Comment:
Do we expect that this will often fail if you're configuring the monitor from outside the proxy the web monitor service is running in? Or more the reverse, that this will succeed but the WM service will have trouble reaching the site?

Reply:
yeah it's a totally valid point, we don't _expect_ major differences afaik but it's certainly possible. that being said, i think _some_  validation that the website (URL) is valid is better than none, and this is the best thing we can do at the moment. any suggestions on how to clarify/improve that?

Reply:
I think this is fine as is, I take it the front end will call this and raising the error is exactly what we're expecting from there? I was just kind of spitballing

Reply:
yup. it's a good thing to callout, we may need to iterate if/when this discrepancy comes up in the wild but i'm good with putting this in and letting that occur 👍